emulator/src/main/java/com/scalar/client/tool/emulator/EmulatorTerminal.java

```diff
@@ -211,4 +215,20 @@ private boolean parseAndRunCommand(String line) {
     }
     return false;
   }
+
+  private String[] getParamsArray(String params, String commandName) {
+    if (commandName.matches("register[\\w-]*")) {
+      return params.split(" ");
+    }
+    List<String> allMatches = new ArrayList<String>();
+    Pattern pattern = Pattern.compile("(\".+?\"|[\\w.-]*)(\\s.+|.*)");
```

Comment:
This pattern can be better explain [here](https://regex101.com/r/7CxhXb/2/tests)...